core/clingutils/CMakeLists.txt

```diff
@@ -61,32 +61,27 @@ foreach(dict ${stldicts})
                                 DEPENDENCIES Core)
 endforeach()
 
-set(CLANG_RESOURCE_DIR_STEM)
-if (builtin_clang)
+if(builtin_clang)
   set(CLANG_RESOURCE_DIR_STEM ${CMAKE_BINARY_DIR}/interpreter/llvm/src/${CMAKE_CFG_INTDIR}/lib/clang)
   set(CLANG_RESOURCE_DIR_VERSION ${LLVM_VERSION})
-else ()
-  set(CLANG_RESOURCE_DIR_STEM ${LLVM_LIBRARY_DIR}/clang)
-  # A user can define a clang version to use, otherwise find it (but will error if more than one version is present)
-  if (NOT DEFINED CLANG_RESOURCE_DIR_VERSION)
-    if (NOT EXISTS ${CLANG_RESOURCE_DIR_STEM})
-      message(FATAL_ERROR "${CLANG_RESOURCE_DIR_STEM} does not exist. Please install clang.")
-    endif()
-    # There is no reasonable way to get the version of clang under which is its resource directory.
-    # For example, lib/clang/5.0.0/include. Deduce it.
-    file(GLOB CHILDREN RELATIVE ${CLANG_RESOURCE_DIR_STEM} ${CLANG_RESOURCE_DIR_STEM}/*)
-    list(LENGTH CHILDREN CHILDREN_LENGTH)
-    if (${CHILDREN_LENGTH} GREATER 1)
-      message(FATAL_ERROR "Found more than one version of clang. CLANG_RESOURCE_DIR_VERSION contains: '${CHILDREN}'." )
-    endif()
-
-    list(GET CHILDREN 0 CLANG_RESOURCE_DIR_VERSION)
+else()
+  execute_process(
+    COMMAND ${LLVM_TOOLS_BINARY_DIR}/clang -print-resource-dir
```

Comment:
`clang` is broken with ROOT-patches; we should not rely on having it. (ROOT doesn't build it either, for good reason.)

Reply:
@amadio I think that's explain why I was not able to generate clang binary for standalone clang/llvm build..

Reply:
The `CLANG_RESOURCE_DIR` variable is only used to copy some system headers into `etc/cling/lib/clang/5.0.x/include`, as shown [here](https://github.com/root-project/root/blob/master/core/clingutils/CMakeLists.txt#L125-L158). We could just move the headers somewhere else to avoid needing this. I think that installing regular headers into `include/root` and headers that currently go into `etc/cling` into `include/cling` or `include/root/cling` (or `include/ROOT/...`) would be better. BTW, this is related to [ROOT-9403](https://sft.its.cern.ch/jira/browse/ROOT-9403), [ROOT-8588](https://sft.its.cern.ch/jira/browse/ROOT-8588), and [ROOT-8589](https://sft.its.cern.ch/jira/browse/ROOT-8589) issues in Jira.

Reply:
We can move the headers some place else but then need to tell clang about the new location.

So - shall we revert this and try your alternative approach? (The clang headers are a runtime thing, they must not go into `include/` but e.g. `share` could work.)

Reply:
You can set `CLANG_RESOURCE_DIR` for clang in CMake, see [here](https://github.com/root-project/root/blob/master/interpreter/llvm/src/tools/clang/CMakeLists.txt#L196-L197). So, sure, I will revert this and redo it without using the clang binary, since the idea is to not have it at all.

Reply:
I would suggest `<libdir>/root/include/...`, for example, `/usr/lib64/root/include/cling` on a system where `<libdir>` is `lib64`.